docs/guides/azure-federation.md

```diff
@@ -0,0 +1,138 @@
+authenticate the AzureRM provider:
+
+```console
+openssl pkcs12 -export -password pass:"Pa55w0rd123" -out client.pfx -inkey client.key -in client.crt
+```
+
+## Step 2: Create Azure Application & Service Principal
+
+Navigate to the Azure portal, and into the Microsoft Entra ID section. Select
+the App Registration blade, and register a new application. Fill in the
+following fields:
+
+- **Name**: this is a friendly identifier and can be anything (e.g. "Coder")
+- **Supported Account Types**: - set to "Accounts in this organizational
+  directory only (single-tenant)"
+
+The **Redirect URI** field does not need to be set in this case. Take note of
+the `Application (client) ID` and `Directory (tenant) ID` values, which will be
+used by Coder.
+
+## Step 3: Assign Client Certificate to the Azure Application
+
+To upload the certificate we created in Step 1, select **Certificates &
+secrets** on the left-hand side, and select **Upload Certificate**. Upload the
+public key file, which is `service-principal.crt` from the example above.
+
+## Step 4: Set Permissions on the Service Principal
+
+Now that the Application is created in Microsoft Entra ID, we need to assign
+permissions to the Service Principal so it can provision Azure resources for
+Coder users. Navigate to the Subscriptions blade in the Azure Portal, select the
+**Subscription > Access Control (IAM) > Add > Add role assignment**.
+
+Set the **Role** that grants the appropriate permissions to create the Azure
+resources you need for your Coder workspaces. `Contributor` will provide
+Read/Write on all Subscription resources. For more information on the available
+roles, see the
+[Microsoft documentation](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles).
+
+## Step 5: Configure Coder to use the Client Certificate
+
+Now that the client certificate is uploaded to Azure, we need to mount the
+certificate files into the Coder deployment. If running Coder on Kubernetes, you
+will need to create the `.pfx` file as a Kubernetes secret, and mount it into
+the Helm chart.
+
+Run the below command to create the secret:
+
+```console
+kubectl create secret generic -n coder azure-client-cert-secret --from-file=client.pfx=/path/to/your/client.pfx
+```
+
+Next, set the following values in Coder's Helm chart:
+
+```yaml
+coder:
+  volumes:
+    - name: "azure-client-cert"
+      secret:
+        secretName: "azure-client-cert-secret"
+  volumeMounts:
+    - name: "azure-client-cert"
+      mountPath: "/home/coder/az/"
+      readOnly: true
+```
+
+Upgrade the Coder deployment using the following `helm` command:
+
+```console
+helm upgrade coder coder-v2/coder -n coder -f values.yaml
+```
+
+## Step 5: Configure your Template to use the Client Certificate
+
+Now that the client certificate is added to Coder, it can be used by Terraform
+to authenticate to Azure. Set the below arguments in your AzureRM provider block
+in the workspace template:
+
+```hcl
```

Comment:
We should recommend Environment variables instead if terraform variables for Secrets.  Then they can use the provider simply as 
```hcl
provider "azurem" {}
```
And add these secrets like
```env
ARM_CLIENT_ID="00000000-0000-0000-0000-000000000000"
ARM_CLIENT_CERTIFICATE_PATH="/path/to/my/client/certificate.pfx"
ARM_CLIENT_CERTIFICATE_PASSWORD="Pa55w0rd123"
ARM_TENANT_ID="10000000-0000-0000-0000-000000000000"
ARM_SUBSCRIPTION_ID="20000000-0000-0000-0000-000000000000"
```
This also doesn't store any secret in coder database as part of the template.

Reply:
@matifali would this be set on the Coder provisioner or server?

Reply:
The Provisioner where terraform runs